### ISSUE TITLE
Android x86_64 compatibility

### ISSUE THREAD

**Issue by 1337CodeGod** [Macoron/whisper.unity#54]
Would it be possible for whisper.cpp to be compiled for Android x86_64 architecture and run using whisper.unity?

I've got a libwhisper.so file built for Android x86_64 and am wondering what my next steps would be to implement it with whisper.unity. Thanks!

**Comment by Macoron** [comment]
I personally never tested that, because I don't have x86 Android device. But you probably can:
1. Compile whisper as a static library .a with x86_64 (see how it's done [in CI pipeline here](https://github.com/Macoron/whisper.unity/blob/8725359a841c20cfbd1c9fe19db1315a4a84c321/.github/workflows/build_cpp.yml#L144))
2. Copy compiled library to package `Plugins/Android` and [replace existing one](https://github.com/Macoron/whisper.unity/tree/8725359a841c20cfbd1c9fe19db1315a4a84c321/Packages/com.whisper.unity/Plugins/Android)
3. [Comment out this code](https://github.com/Macoron/whisper.unity/blob/8725359a841c20cfbd1c9fe19db1315a4a84c321/Packages/com.whisper.unity/Editor/CheckAndroidPreprocessBuild.cs#L10) which checks `PlayerSettings`. I added that just as a warning and it won't break anything.
4. In `PlayerSettings` enable x86_64 architecture

Should probably work after that. Please write back if it does.

P.S. For what target do you compile x86_64? Is that some Chrome OS IoT stuff?

**Comment by 1337CodeGod** [comment]
Thanks for the directions, I'll give it a try! My target device is the Magic Leap 2

**Comment by 1337CodeGod** [comment]
The app builds and starts after replacing the libwhisper.a with one compiled for Android x86_64, but crashes immediately at runtime.

**Comment by Babilinski** [comment]
Heads-up you may need to update the build.sh script that is in the root folder to respect the architecture specified in the build pipeline.

**Comment by Macoron** [comment]
> Heads-up to anyone who stubbles upon this, you may need to update the build.sh script that is in the root folder to respect the architecture specified in the build pipeline.
> 
> https://github.com/Macoron/whisper.unity/blob/master/build_cpp.sh#L78

Does the compiled library work on your target x86_64 device?

**Comment by Babilinski** [comment]
Yes, I just pushed up the changes and configured the project in my fork for android x86_64:
https://github.com/Babilinski/whisper.unity

**Comment by bariskes11** [comment]
I also need that but I'm using windows can any one help  with library files it never load's to Magic Leap 2 device like that at  the moment..
